java/calcite/src/main/java/com/mapd/parser/server/CalciteServerHandler.java

```diff
@@ -175,10 +182,29 @@ public TPlanResult process(String user,
       Pair<String, SqlIdentifierCapturer> res;
       SqlNode node;
       try {
-        MapDParserOptions parserOptions = new MapDParserOptions(
-                filterPushDownInfo, legacySyntax, isExplain, isViewOptimize);
-        res = parser.process(sqlText, parserOptions);
-        jsonResult = res.left;
+        if (!isRAQuery) {
+          MapDParserOptions parserOptions = new MapDParserOptions(filterPushDownInfo, legacySyntax, isExplain,
+              isViewOptimize);
+          res = parser.process(queryText, parserOptions);
+          jsonResult = res.left;
+          capturer = res.right;
+
+          primaryAccessedObjects.tables_selected_from = new ArrayList<>(capturer.selects);
+          primaryAccessedObjects.tables_inserted_into = new ArrayList<>(capturer.inserts);
+          primaryAccessedObjects.tables_updated_in = new ArrayList<>(capturer.updates);
+          primaryAccessedObjects.tables_deleted_from = new ArrayList<>(capturer.deletes);
+
+          // also resolve all the views in the select part
+          // resolution of the other parts is not
+          // necessary as these cannot be views
+          resolvedAccessedObjects.tables_selected_from = new ArrayList<>(parser.resolveSelectIdentifiers(capturer));
+          resolvedAccessedObjects.tables_inserted_into = new ArrayList<>(capturer.inserts);
+          resolvedAccessedObjects.tables_updated_in = new ArrayList<>(capturer.updates);
+          resolvedAccessedObjects.tables_deleted_from = new ArrayList<>(capturer.deletes);
+
+        } else {
+          jsonResult = parser.optimizeRAQuery(queryText);
```

Comment:
1) Why RAQuery does not need to specify `primaryAccessedObjects` and `resolvedAccessedObjects`?

2) Also, if RAQuery only needs to apply some specific optimization rules, then can we completely separate the logic to keep the current OmniSci Calcite server logic as stable as possible to make the least change?

Like..

if (isRAQuery) {
    jsonResult = parser.optimizeRAQuery(queryText);
} else {
  // put the existing OmniSci parsing logic --
try {
       ...
       try {
         ...
       } catch(...) { ... }
  } catch(...) { ... }
}

Reply:
1)I'm not sure about `primaryAccessedObjects` and `resolvedAccessedObjects`. I don't really understand what are they for. Also, we didn't found any case, where our query does not work without them. And it looks like they come from SQLNode, and I don't really see the easy way to extract them from calcite ast. 

2) I see, what you mean, will send a fix soon

Reply:
I've sent a commit with possible fix of 2